tests/block_builder_test.rs

```diff
@@ -1,72 +1,195 @@
+        .into();
+
+    // Make a bundle out of them
+    let bundle = SignetEthBundle {
+        bundle: EthSendBundle {
+            replacement_uuid: Some("test-replacement-uuid".to_string()),
+            txs: vec![ru_txn],
+            block_number: sim_env.rollup_block_number(),
+            ..Default::default()
+        },
+        host_txs: vec![host_txn],
+    };
+
+    // Add it to the sim cache
+    let sim_items = SimCache::new();
+    sim_items.add_bundle(bundle, 7).expect("adds bundle");
+
+    // Setup the simulator environment
+    let (submit_tx, mut submit_rx) = unbounded_channel();
+    let simulator_task = SimulatorTask::new(sim_rx, host_provider, ru_provider);
+    let simulator_jh = simulator_task.spawn_simulator_task(sim_items, submit_tx);
+
+    // Send a new environment to tick the block builder simulation loop off
+    sim_tx.send(Some(sim_env)).unwrap();
+
+    // Wait for a result and assert on it
+    let got = submit_rx.recv().await.expect("built block");
+    dbg!(&got.block);
+    assert_eq!(got.block.transactions().len(), 1);
+    assert_eq!(got.block.host_transactions().len(), 1);
+
+    // Cleanup
+    simulator_jh.abort();
+}
+
+async fn latest_sim_env(
+    env_task: &EnvTask,
+    host_provider: &HostProvider,
+    ru_provider: &RuProvider,
+) -> SimEnv {
+    let host_previous =
+        host_provider.get_block(BlockId::latest()).await.unwrap().unwrap().header.inner;
+    let ru_previous =
+        ru_provider.get_block(BlockId::latest()).await.unwrap().unwrap().header.inner;
+
+    let host_env = env_task.construct_host_env(host_previous);
+    let ru_env = env_task.construct_rollup_env(ru_previous);
+
+    SimEnv { host: host_env, rollup: ru_env, span: tracing::Span::none() }
+}
+
+fn spawn_host_anvil() -> (AnvilInstance, HostProvider, PrivateKeySigner) {
+    let anvil = Anvil::new().chain_id(constants().host_chain_id()).spawn();
+    let key = anvil.keys()[0].clone();
+    let signer = PrivateKeySigner::from_bytes(&B256::from_slice(&key.to_bytes())).unwrap();
+    let wallet = anvil.wallet().expect("anvil wallet");
+    let provider = ProviderBuilder::new_with_network()
+        .disable_recommended_fillers()
+        .filler(BlobGasFiller)
+        .with_gas_estimation()
+        .with_nonce_management(SimpleNonceManager::default())
+        .fetch_chain_id()
+        .wallet(wallet)
+        .connect_http(anvil.endpoint_url());
+
+    (anvil, provider, signer)
+}
+
+fn spawn_rollup_anvil() -> (AnvilInstance, RuProvider, Vec<PrivateKeySigner>) {
+    let anvil = Anvil::new().chain_id(constants().ru_chain_id()).spawn();
+    let signers: Vec<_> = anvil
+        .keys()
+        .iter()
+        .take(2)
+        .map(|key| PrivateKeySigner::from_bytes(&B256::from_slice(&key.to_bytes())).unwrap())
+        .collect();
+    assert!(signers.len() >= 2, "rollup anvil must provide at least two accounts");
+    let provider = RootProvider::new_http(anvil.endpoint_url());
+
+    (anvil, provider, signers)
+}
```

Comment:
NB: These are intentionally not forked off of RPC so that they don't rely on Pecorino state at all yet. These are simply simulating "dumb" Signet bundles. The next step to shore these up is going to be forking them off of the actual host and rollup RPCs (once Pecorino is running again) and then making these "smart" bundles that actually call Signet contracts.